docs/reference/query-dsl/regexp-syntax.asciidoc

```diff
@@ -37,7 +37,55 @@ backslash or surround it with double quotes. For example:
 \\                  # renders as a literal '\'
 "john@smith.com"    # renders as 'john@smith.com'
 ....
-    
+
+[NOTE]
+====
+
+The backslash character is an escape character in both JSON strings and regular
+expressions. You need to escape both backslashes in a query, unless you use a
+language client, which takes care of this. For example, the keyword string `a\b`
+needs to be indexed as `"a\\b"`:
+
+////
+[source,console]
+----
+PUT my-index-000001
```

Comment:
I'm not sure the mapping here should be hidden -- if you try to reproduce this without using the same mapping (i.e. with `my_field` either implicitly or explicitly as `"type": "text"`) then the example doesn't quite work.

I defer to others, though, this is outside my wheelhouse.

Reply:
@joegallo I like your suggestion to simplify the example by removing the mapping and querying `my_field.keyword` instead. I've updated the example accordingly.